app/assets/javascripts/compute_resource.js

```diff
@@ -252,3 +252,26 @@ function vsphereGetResourcePools(item) {
     }
   })
 }
+
+function vsphereStoragePodSelected(item) {
+  selected = $(item).val();
+  dropdown = $(item).closest('select');
+  datastore = $(item).parentsUntil('.fields').parent().find('#host_compute_attributes_volumes_attributes_0_datastore');
```

Comment:
How about matching a regex instead of finding an ID directly? I understand the `0` is the problem right?

Also would this work on a separate file? It's part of compute_resource but I think it's a bit easier to understand conceptually if we stick all the VMware logic in its own file, 😶